app/jobs/solidus_virtual_gift_card/void_expired_authorized_events_job.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+module SolidusVirtualGiftCard
+  class VoidExpiredAuthorizedEventsJob < ApplicationJob
+    queue_as :default
+
+    def perform
+      # Subquery to get the latest 'created_at' timestamp for each virtual_gift_card_event_id
+      subquery = ::Spree::VirtualGiftCardEvent
+                 .group(:virtual_gift_card_id)
+                 .select('virtual_gift_card_id, MAX(created_at) AS max_created_at')
```

Comment:
I think we can filter out events here with the same timestamp to reduce the subquery size as you have done in the main query:

`Filters events where the 'created_at' is older than the configured 'authorize_timeout'`